File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#define UPPER 65
#define LOWER 97
#define ATOZ 26

int main (int argc, string argv[])
{
    //Check that only two command line arguments were entered by the user
    if (argc == 2)
    {
        //Convert key entered from string to integer
        int k = atoi(argv[1]);

        //Request plaintext from user
        printf("plaintext: ");
        string p = get_string();

        //Get length of plaintext string and create a char to hold each letter of ciphertext
        int len = strlen(p);
        char c;

        //Print output
        printf("ciphertext: ");

        //Iterate through length of plaintext string, char by char
        for (int i = 0; i < len; i++)
        {
            //Check if char is alphabetical
            if (isalpha(p[i]))
            {
                //Check case of char to maintain it
                if (isupper(p[i]))
                {
                    //Convert char to alphabetical index, rotate by key, covert back to ASCII index, print char
                    c = (((p[i] - UPPER) + k) % ATOZ) + UPPER;
                    printf("%c", c);
                }
                else
                {
                    c = (((p[i] - LOWER) + k) % ATOZ) + LOWER;
                    printf("%c", c);
                }
            }
            //If char is another type of character, print as is
            else
            {
                c = p[i];
                printf("%c", c);
            }
        }
        //Print new line after exiting loop that prints ciphertext
        printf("\n");

        return 0;
    }
    //If the incorrect number of command line arguments were entered by user, return error
    else
    {
        return 1;
    }
}

File: caesar.py
import cs50
import sys

def main():
    # alphabetic ASCII constants
    upper = 65
    lower = 97
    atoz = 26

    # check that two command line arguments were entered
    if len(sys.argv) == 2:

        # convert key entered from string to integer
        k = int(sys.argv[1])

        # request plaintext string from user
        print("plaintext: ", end="")
        p = cs50.get_string()

        # start printing output
        print("ciphertext: ", end="")

        # iterate through each char in plaintext string
        for c in p:

            # check if char is alphabetic
            if c.isalpha():

                # obtain numerical ASCII value of char
                num = ord(c)

                # check case of char to maintain it
                if c.isupper():
                    # rotate uppercase char by key and print
                    num = (((num - upper) + k) % atoz) + upper
                    print(chr(num), end="")

                else:
                    # rotate lowercase char by key and print
                    num = (((num - lower) + k) % atoz) + lower
                    print(chr(num), end="")

            else:
                # print non-alphabetic chars as they are
                print(c, end="")

        print()
        return 0

    # return error message if incorrect number of command line arguments entered
    else:
        print("Usage: python caesar.py k")
        return 1

if __name__ == "__main__":
    main()